src/main/common/filter.c

```diff
@@ -309,3 +309,23 @@ void simpleLPFilterInit(simpleLowpassFilter_t *filter, int32_t beta, int32_t fpS
     filter->beta = beta;
     filter->fpShift = fpShift;
 }
+
+uint16_t updateMovingAverageUint16(movingAverageStateUint16_t *state, uint16_t newValue)
+{
+    state->sum -= state->values[state->pos];
+    state->sum += newValue;
+    state->values[state->pos] = newValue;
+    state->pos = (state->pos + 1) % state->size;
+
+    return state->sum / state->size;
```

Comment:
does this need `lrintf` also like line 330 or the other way around?

Reply:
No because `sum` and `size` are both integers, The solution will be integer as well.